File: backend/app/schemas/outreach.py
"""Outreach schemas."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel

from app.models.enums import OutreachStatus, OutreachTone
from app.schemas.common import TimestampedModel


class OutreachRead(TimestampedModel):
    id: str
    scan_id: str
    account_id: str
    person_id: str | None = None
    subject: str
    body: str
    tone: OutreachTone
    status: OutreachStatus
    guardrail_notes_json: list[Any] | None = None
    reviewer_feedback: str | None = None


class OutreachList(BaseModel):
    items: list[OutreachRead]
    total: int


class OutreachReject(BaseModel):
    feedback: str | None = None


class OutreachPatch(BaseModel):
    subject: str | None = None
    body: str | None = None
    tone: OutreachTone | None = None

File: frontend/lib/types/outreach.ts
// Source: backend/app/schemas/outreach.py and backend/app/models/enums.py

import type { Timestamped } from "./common";

export const OUTREACH_TONES = ["warm", "technical", "executive", "concise"] as const;
export type OutreachTone = (typeof OUTREACH_TONES)[number];

export const OUTREACH_STATUSES = [
  "pending_review",
  "approved",
  "rejected",
  "edited",
] as const;
export type OutreachStatus = (typeof OUTREACH_STATUSES)[number];

export type OutreachRead = Timestamped & {
  id: string;
  scan_id: string;
  account_id: string;
  person_id?: string | null;
  subject: string;
  body: string;
  tone: OutreachTone;
  status: OutreachStatus;
  guardrail_notes_json?: unknown[] | null;
  reviewer_feedback?: string | null;
};

export type OutreachList = {
  items: OutreachRead[];
  total: number;
};

export type OutreachReject = {
  feedback?: string;
};

export type OutreachPatch = {
  subject?: string;
  body?: string;
  tone?: OutreachTone;
};
